integrated_channels/integrated_channel/models.py

```diff
@@ -62,3 +77,55 @@ class EnterpriseCustomerPluginConfiguration(TimeStampedModel):
 
     class Meta:
         abstract = True
+
+    def transmit_learner_data(self):
+        """
+        Collect and transmit learner data for the enterprise_customer.
+        """
+        for learner_data in self.collect_learner_data():
+            LOGGER.info(learner_data)
+
+            # TODO ENT-221 will handle transmitting the learner data to the channel
+
+    def collect_learner_data(self):
+        """
+        Collect learner data for the enterprise_customer where data sharing consent is granted.
+
+        Yields a dict for each enrollment, containing:
+
+        * enterprise_enrollment: EnterpriseCourseEnrollment
+        * certificate: GeneratedCertificate for the user+course, None if not found.
+          "Course completion" occurs when course certificates are issued.
+        """
+
+        # Fetch the consenting enrollment data, including the enterprise_customer_user
+        enrollment_queryset = EnterpriseCourseEnrollment.objects.select_related(
+            'enterprise_customer_user'
+        ).filter(
+            consent_granted=True,
+            enterprise_customer_user__enterprise_customer=self.enterprise_customer,
+        )
+        for enterprise_enrollment in enrollment_queryset:
+
+            if CourseKey is None:
+                raise NotConnectedToEdX("This package must be installed in an OpenEdX environment.")
+
+            # Fetch the user+course certificate, if found
+            try:
+                certificate = GeneratedCertificate.eligible_certificates.get(
+                    user__id=enterprise_enrollment.enterprise_customer_user.user_id,
+                    course_id=CourseKey.from_string(enterprise_enrollment.course_id),
+                )
+            except GeneratedCertificate.DoesNotExist:
+                certificate = None
+
+            yield self.learner_data_record(
+                enterprise_enrollment=enterprise_enrollment,
+                certificate=certificate,
+            )
+
+    def learner_data_record(self, enterprise_enrollment, certificate=None):
```

Comment:
37299d0 changes this to [`get_learner_data()`](https://github.com/edx/edx-enterprise/pull/59/commits/37299d0df0e34dd0e9f00da8bfacee8900f2561e#diff-4386ccd909983c900b04b5a992e66fddR127), which returns a `LearnerDataTransmissionAudit`

`LearnerDataTransmissionAudit` now has a `payload()` method, which does the serialization.